service/resource/certificate/update.go

```diff
@@ -41,13 +42,38 @@ func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState inte
 		return microerror.Mask(err)
 	}
 
-	for _, certificateFile := range updateCertificateFiles {
-		r.logger.Log("debug", fmt.Sprintf("writing certificate: %s", certificateFile.path))
-		if err := afero.WriteFile(r.fs, certificateFile.path, []byte(certificateFile.data), r.certificatePermission); err != nil {
+	// Write the update state certificate files.
+	for _, fileToWrite := range updateCertificateFiles {
+		r.logger.Log("debug", fmt.Sprintf("writing certificate: %s", fileToWrite.path))
+		if err := afero.WriteFile(r.fs, fileToWrite.path, []byte(fileToWrite.data), r.certificatePermission); err != nil {
```

Comment:
This makes sense to me. The state necessary to do the update is just injected and then plainly used as it is.